// csstree-validator/cjs/helpers.cjs
'use strict';

const fs = require('fs');
const path = require('path');
const validate = require('./validate.cjs');

function createResult() {
    const result = Object.create(null);

    result[Symbol.iterator] = function*() {
        for (const [filename, errors] of Object.entries(this)) {
            yield [filename, errors];
        }
    };

    return result;
}

function defaultShouldBeValidated(filename) {
    return path.extname(filename) === '.css';
}

function collectFiles(testPath, shouldBeValidated) {
    try {
        if (fs.statSync(testPath).isDirectory()) {
            return [].concat(...fs.readdirSync(testPath).map(dirFilename =>
                collectFiles(path.join(testPath, dirFilename), shouldBeValidated)
            )).sort();
        } else {
            return shouldBeValidated(testPath) ? [testPath] : [];
        }
    } catch (e) {
        return [testPath];
    }
}

function validateDictionary(dictionary) {
    const result = createResult();

    for (const filename of Object.keys(dictionary).sort()) {
        result[filename] = validate.validate(dictionary[filename], filename);
    }

    return result;
}

function validateString(css, filename) {
    const result = createResult();

    if (!filename) {
        filename = '<unknown>';
    }

    result[filename] = validate.validate(css, filename);

    return result;
}

function validateFile(filename) {
    const result = createResult();
    let css;

    try {
        css = fs.readFileSync(filename, 'utf-8');
        result[filename] = validate.validate(css, filename);
    } catch (e) {
        result[filename] = [e];
    }

    return result;
}

function validateFileList(list) {
    const result = createResult();

    for (const filename of list) {
        const res = validateFile(filename)[filename];

        if (res && res.length !== 0) {
            result[filename] = res;
        }
    }

    return result;
}

function validatePathList(pathList, filter) {
    if (typeof filter !== 'function') {
        filter = defaultShouldBeValidated;
    }

    const fileList = new Set([].concat(...pathList.map(path =>
        collectFiles(path, filter)
    )));

    return validateFileList([...fileList].sort());
}

function validatePath(searchPath, filter) {
    if (typeof filter !== 'function') {
        filter = defaultShouldBeValidated;
    }

    return validateFileList(collectFiles(searchPath, filter));
}

exports.validateDictionary = validateDictionary;
exports.validateFile = validateFile;
exports.validatePath = validatePath;
exports.validatePathList = validatePathList;
exports.validateString = validateString;
